### ISSUE TITLE
DEV: Spread integration tests across multiple domains

### ISSUE THREAD

**Issue by tlimoncelli** [StackExchange/dnscontrol#1744]
Some integration tests take 7 or more minutes. Others run very quickly.  The slow tests delay the development process.

I've tried to optimize the tests as much as possible but certain slow-pokes like AZURE_DNS do all DNS updates one API call at a time, thus none of the things I've tried have made much improvement.

I think the only way to make the integration tests faster for AZURE_DNS, GANDI_V5, and CLOUDFLARE_API is to divide the work across multiple domains and do them in parallel.

## Ideal version:

Specifying multiple domains (for example `AZURE_DNS_DOMAIN=dnscontrol-azure.com,dnscontrol-azure2.com`) would divided the tests evenly across the two domains. 

Ideally one domain would get the even tests and the other would get the odd tests. This would load balance between the domains better because the last 20+ tests are frequently no-ops.

## Quick version

We could simply use the `-start` and `-end` flags to allocate the tests directly in the CircleCI pipeline.

**Comment by tlimoncelli** [comment]
Won't do.
